src/python3/CMakeLists.txt

```diff
@@ -11,3 +11,14 @@ set_target_properties(cpp
 )
 # this is on by default "pybind11_add_module" but can interfere with coverage so we disable it if coverage is enabled
 set_property(TARGET cpp PROPERTY INTERPROCEDURAL_OPTIMIZATION ${PHARE_INTERPROCEDURAL_OPTIMIZATION})
+
+if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+  pybind11_add_module(cpp_dbg cpp_simulator.cpp)
+  target_link_libraries(cpp_dbg PUBLIC phare_simulator)
+  target_compile_options(cpp_dbg PRIVATE ${PHARE_FLAGS} -DPHARE_HAS_HIGHFIVE=${PHARE_HAS_HIGHFIVE} -DPHARE_DIAG_DOUBLES=1 -DPHARE_CPP_MOD_NAME=cpp_dbg)
```

Comment:
"-DPHARE_DIAG_DOUBLES=1" should probably be "-DPHARE_DEBUG_MODE=1"

and either "PHARE_DIAG_DOUBLES" is defined to true/1 if PHARE_DEBUG_MODE is active, or drop PHARE_DIAG_DOUBLES entirely and just use PHARE_DEBUG_MODE to pick between float/double in h5writer.h "using FloatType = std::conditional_t<PHARE_DIAG_DOUBLES, double, float>;"

Reply:
I think we might, for some reason, want to have the option of using double diags beyond debug mode so I'd keep that option, and yes keep setting it to true when in debug also, is useful

Reply:
"__debug__" is false when exectuting with "python -O script.py"